### DISCUSSION TITLE
Problem with creation of a database

### DISCUSSION THREAD

**Discussion by VilobatirKedis** [Budibase/budibase#2267] [Help]
Hello,

I have a query that creates a database and it has a parameter for the name of the database, like this:

![query](https://user-images.githubusercontent.com/81375075/128208476-b71c2e62-bfc2-44d6-b056-722cd93a9046.PNG)

I am using this query as an action on a button that gets the name from a text field in a form, like this:

![button](https://user-images.githubusercontent.com/81375075/128208818-61104416-1033-45fb-95ec-5e2a401d84ce.PNG)

However, when I launch the application, input a name and then press the button, it gives me an error:

![Cattura](https://user-images.githubusercontent.com/81375075/128209235-97846b84-bfde-4590-876b-fbfc72809679.PNG)

The query itself works because I tested it with a normal string value (like "Hello") and it creates the database
as I wanted so the problem comes when I try to use the text written by the user.

**Comment by oswaldB** [comment]
And if your user text is Hello, is it working ? If not, maybe "Hello" with the "?

**Reply by VilobatirKedis** [reply]
I haven't explained myself well. If I put Hello (without quotation marks) directly inside the parameter value  in the button action (hardcoded) it works

**Comment by mjashanks** [comment]
Seems like it should work. Do the database credentials you are using have permission to `CREATE DATABASE` ?

Could you try another query e.g. `SELECT` ? If this was successful, that would suggest that you just aren't allowed to `CREATE DATABASE` in the current scope.

Not sure what DB you are using, but when I setup my postgres connection in Budibase, one of the fields is `database`. It wouldn't surprise me if it failed to create a database, when connected to another database (guessing here).

**Reply by VilobatirKedis** [reply]
I am using a MySQL DB and yes, it has permission to `CREATE DATABASE` because if I hardcode a plain string directly in the parameter value it works

**Comment by shogunpurple** [comment]
@VilobatirKedis would you be able to show the result of that request in the network tab? It would be helpful to see the exact error that is coming back from the database/server

**Comment by VilobatirKedis** [comment]
As suggested by @shogunpurple, I looked for the result in the network tab and this happens:

![Cattura](https://user-images.githubusercontent.com/81375075/128300559-e5d1a981-9517-41fc-bce4-0b2f533c9298.PNG)

So he is trying to create a database with an empty name even if I input something in the text field in the form, so
`{{NewForm.Fields.Text}}` is wrong. Is the syntax wrong or do I need to access another property to get the right value from the form?

**Reply by VilobatirKedis** [reply]
Nevermind, I just found out the problem. The button needs to be included in the form to reference other fields, that's why the value was blank.

**Reply by shogunpurple** [reply]
Thanks for updating us @VilobatirKedis - that is actually something we've chatted about this week. It's not immediately obvious that you need a form to use input values.
